mbed_host_tests/host_tests_conn_proxy/conn_proxy.py

```diff
@@ -81,46 +81,35 @@ def conn_primitive_factory(conn_resource, config, event_queue, logger):
     @param logger Host Test logger instance
     @return Object of type <ConnectorPrimitive> or None if type of connection primitive unknown (conn_resource)
     """
+    polling_timeout = int(config.get('polling_timeout', 60))
+    logger.prn_inf("notify event queue about extra %d sec timeout for serial port pooling"%polling_timeout)
+    event_queue.put(('__timeout', polling_timeout, time()))
+
     if conn_resource == 'serial':
         # Standard serial port connection
         # Notify event queue we will wait additional time for serial port to be ready
 
         # Get extra configuration related to serial port
         port = config.get('port')
         baudrate = config.get('baudrate')
-        serial_pooling = int(config.get('serial_pooling', 60))
-
-        logger.prn_inf("notify event queue about extra %d sec timeout for serial port pooling"%serial_pooling)
-        event_queue.put(('__timeout', serial_pooling, time()))
 
         logger.prn_inf("initializing serial port listener... ")
         connector = SerialConnectorPrimitive(
             'SERI',
             port,
             baudrate,
             config=config)
-        return connector
-
     elif conn_resource == 'grm':
         # Start GRM (Gloabal Resource Mgr) collection
-
-        # Get extra configuration related to remote host
-        remote_pooling = int(config.get('remote_pooling', 30))
-
-        # Adding extra timeout for connection to remote resource host
-        logger.prn_inf("notify event queue about extra %d sec timeout for remote connection"%remote_pooling)
-        event_queue.put(('__timeout', remote_pooling, time()))
-
         logger.prn_inf("initializing global resource mgr listener... ")
         connector = RemoteConnectorPrimitive(
             'GLRM',
             config=config)
-        return connector
-
     else:
         logger.pn_err("unknown connection resource!")
         raise NotImplementedError("ConnectorPrimitive factory: unknown connection resource '%s'!"% conn_resource)
-        return None
```

Comment:
Not sure how Python will react to this, since the `connector` variable hasn't been referenced at this point. Maybe we should add an explicit `connector = None` at the top before all the `if conn_resource ==` checks?